psh/ifconfig/ifconfig.c

```diff
@@ -0,0 +1,527 @@
+/*
+ * Phoenix-RTOS
+ *
+ * ifconfig - configure network interfaces
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Andrzej Stalke
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+
+#include <inttypes.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+
+#include "../psh.h"
+
+#define IFCONFIG_VERBOSE 1
+#define IFCONFIG_ALL     2
+#define IFCONFIG_HELP    4
+
+enum operation {operation_setFlag, operation_unsetFlag, operation_toggleFlag};
+
+static void psh_ifconfigInfo(void)
+{
+	printf("configures network interfaces");
+}
+
+
+static inline const char *psh_ifconfigPrintFlag(unsigned int flag)
+{
+	switch (flag) {
+		case IFF_UP:
+			return "UP";
+		case IFF_BROADCAST:
+			return "BROADCAST";
+		case IFF_DEBUG:
+			return "DEBUG";
+		case IFF_LOOPBACK:
+			return "LOOPBACK";
+		case IFF_POINTOPOINT:
+			return "POINTOPOINT";
+		case IFF_NOTRAILERS:
+			return "NOTRAILERS";
+		case IFF_RUNNING:
+			return "RUNNING";
+		case IFF_NOARP:
+			return "NOARP";
+		case IFF_PROMISC:
+			return "PROMISC";
+		case IFF_ALLMULTI:
+			return "ALLMULTI";
+		case IFF_OACTIVE:
+			return "OACTIVE";
+		case IFF_SIMPLEX:
+			return "SIMPLEX";
+		case IFF_LINK0:
+			return "LINK0";
+		case IFF_LINK1:
+			return "LINK1";
+		case IFF_MULTICAST:
+			return "MULTICAST";
+		case IFF_DYNAMIC:
+			return "DYNAMIC";
+		default:
+			return "UNKNOWN_FLAG";
+	}
+}
+
+
+static inline int psh_ifconfigPrintInterface(const struct ifaddrs *interface, int sd)
+{
+	unsigned int flags, i;
+	struct in_addr addr, broadcast, mask;
+	int ret;
+	struct ifreq ioctlInterface;
+
+	(void)strncpy(ioctlInterface.ifr_name, interface->ifa_name, IFNAMSIZ - 1);
+	/* Get interface flags */
+	ret = ioctl(sd, SIOCGIFFLAGS, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFFLAGS)");
+		return ret;
+	}
+	flags = ioctlInterface.ifr_flags;
+
+	/* Get interface IP address */
+	ret = ioctl(sd, SIOCGIFADDR, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFADDR)");
+		return ret;
+	}
+	addr = ((struct sockaddr_in *)&ioctlInterface.ifr_addr)->sin_addr;
+
+	/* Get interface broadcast IP address */
+	ret = ioctl(sd, SIOCGIFBRDADDR, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFBRDADDR)");
```

Comment:
I've noticed issue running this PR. for lwip running only ppp and loopback interfaces, ifconfig fails in this line, like on screenshot below:
![2023-11-22-193841_619x442_scrot](https://github.com/phoenix-rtos/phoenix-rtos-utils/assets/141153/fd078bbc-95dd-422f-9621-a7bf08b58b49)
ifconfig should list at least `lo0` interface as it is same on ia32-generic, I believe ifconfig should support point to point interfaces too. Directly reading /dev/ifstatus shows all available interfaces correctly.

Reply:
Now works as expected.